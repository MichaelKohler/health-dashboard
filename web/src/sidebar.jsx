import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Divider from '@material-ui/core/Divider';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemIcon from '@material-ui/core/ListItemIcon';
import ListItemText from '@material-ui/core/ListItemText';
import ListSubheader from '@material-ui/core/ListSubheader';
import DashboardIcon from '@material-ui/icons/Dashboard';
import FitnessCenterIcon from '@material-ui/icons/FitnessCenter';
import SmokingRoomsIcon from '@material-ui/icons/SmokingRooms';
import LockIcon from '@material-ui/icons/Lock';
import LockOpenIcon from '@material-ui/icons/LockOpen';
import DirectionsRunIcon from '@material-ui/icons/DirectionsRun';

// The usage of React.forwardRef will no longer be required for react-router-dom v6.
// see https://github.com/ReactTraining/react-router/issues/6056
const AdapterLink = React.forwardRef((props, ref) => <Link innerRef={ ref } { ...props }/>); // eslint-disable-line react/display-name

const mapStateToProps = (state) => state;

// eslint-disable-next-line react/no-multi-comp
export function Sidebar(props) {
  return (
      <List>
          <ListItem button component={ AdapterLink } to="/">
              <ListItemIcon>
                  <DashboardIcon/>
              </ListItemIcon>
              <ListItemText primary="Dashboard"/>
          </ListItem>
          { !props.isLoggedIn && (
              <ListItem button component={ AdapterLink } to="/login">
                  <ListItemIcon>
                      <LockIcon/>
                  </ListItemIcon>
                  <ListItemText primary="Login"/>
              </ListItem>
          ) }
          { props.isLoggedIn && (
              <ListItem button component={ AdapterLink } to="/logout">
                  <ListItemIcon>
                      <LockOpenIcon/>
                  </ListItemIcon>
                  <ListItemText primary="Logout"/>
              </ListItem>
        ) }

          <Divider/>
          <ListSubheader inset>Reports</ListSubheader>

          <ListItem button component={ AdapterLink } to="/cigarettes">
              <ListItemIcon>
                  <SmokingRoomsIcon/>
              </ListItemIcon>
              <ListItemText primary="Cigarettes"/>
          </ListItem>
          <ListItem button component={ AdapterLink } to="/weight">
              <ListItemIcon>
                  <FitnessCenterIcon/>
              </ListItemIcon>
              <ListItemText primary="Weight"/>
          </ListItem>
          <ListItem button component={ AdapterLink } to="/stairs">
              <ListItemIcon>
                  <DirectionsRunIcon/>
              </ListItemIcon>
              <ListItemText primary="Stairs"/>
          </ListItem>
      </List>
  );
}

Sidebar.propTypes = {
  isLoggedIn: PropTypes.bool,
};

export default connect(
  mapStateToProps,
)(Sidebar);
