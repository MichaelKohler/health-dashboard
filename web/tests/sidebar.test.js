/* global it, expect */
/* eslint-disable react/jsx-filename-extension */

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Sidebar } from '../src/sidebar.jsx';

it('renders correctly', () => {
  const tree = renderer
    .create(<MemoryRouter><Sidebar/></MemoryRouter>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* eslint-enable react/jsx-filename-extension */
