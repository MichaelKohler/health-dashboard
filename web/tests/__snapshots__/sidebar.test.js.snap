// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 13h8V3H3v10zm0 8h8v-6H3v6zm10 0h8V11h-8v10zm0-18v6h8V3h-8z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Dashboard
      </span>
    </div>
  </a>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/login"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M18 8h-1V6c0-2.76-2.24-5-5-5S7 3.24 7 6v2H6c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2zm-6 9c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2zm3.1-9H8.9V6c0-1.71 1.39-3.1 3.1-3.1 1.71 0 3.1 1.39 3.1 3.1v2z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Login
      </span>
    </div>
  </a>
  <hr
    className="MuiDivider-root"
  />
  <li
    className="MuiListSubheader-root MuiListSubheader-inset MuiListSubheader-sticky MuiListSubheader-gutters"
  >
    Reports
  </li>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/cigarettes"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M2 16h15v3H2zm18.5 0H22v3h-1.5zM18 16h1.5v3H18zm.85-8.27c.62-.61 1-1.45 1-2.38C19.85 3.5 18.35 2 16.5 2v1.5c1.02 0 1.85.83 1.85 1.85S17.52 7.2 16.5 7.2v1.5c2.24 0 4 1.83 4 4.07V15H22v-2.24c0-2.22-1.28-4.14-3.15-5.03zm-2.82 2.47H14.5c-1.02 0-1.85-.98-1.85-2s.83-1.75 1.85-1.75v-1.5c-1.85 0-3.35 1.5-3.35 3.35s1.5 3.35 3.35 3.35h1.53c1.05 0 1.97.74 1.97 2.05V15h1.5v-1.64c0-1.81-1.6-3.16-3.47-3.16z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Cigarettes
      </span>
    </div>
  </a>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
    href="/weight"
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    role="button"
    tabIndex={0}
  >
    <div
      className="MuiListItemIcon-root"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.57 14.86L22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
        />
      </svg>
    </div>
    <div
      className="MuiListItemText-root"
    >
      <span
        className="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
      >
        Weight
      </span>
    </div>
  </a>
</ul>
`;
